### PR TITLE
Add lambda and script to dump v3 account data

### PR DESCRIPTION
**Pull request by alex-chew**
*Description of changes:*
This PR adds a lambda function (and corresponding script) to dump account data from CustomersTable and from the CognitoUserPool, which will allow Portal owners to migrate their users if they need to change their UserPool in ways which require replacement. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lambdas/dump-v3-account-data/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * set to the username as specified in the `usernamesByUserId` Map.
+ */
+const insertUsernames = ({ accounts, usernamesByUserId }) => accounts
+  .map(account => ({
+    ...account,
+    username: usernamesByUserId.get(account.userPoolId),
+  }))
+
+/**
+ * Formats an account account object as a string of tab-separated values.
```
Comment (Trial-In-Error) [inline lambdas/dump-v3-account-data/index.js]: account account?

File: cloudformation/template.yaml [hunk 2]
```diff
@@ -1476,6 +1476,61 @@ Resources:
...
             HostedZoneId: 'Z2FDTNDATAQYW2'
 
+  DumpV3AccountDataFn:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ../lambdas/dump-v3-account-data
+      Handler: index.handler
+      MemorySize: 512
+      Role: !GetAtt DumpV3AccountDataExecutionRole.Arn
+      Runtime: nodejs8.10
```
[the remark this answers is not in the document]
Comment (alex-chew) [reply cloudformation/template.yaml]: Changed.

File: lambdas/common-layer/nodejs/node_modules/dev-portal-common/pager.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const fetchAllItems = async ({ fetchPage, commonParams, selectItems, getNextPageParams }) => {
```
[the remark this answers is not in the document]
Comment (alex-chew) [reply lambdas/common-layer/nodejs/node_modules/dev-portal-common/pager.js]: I've also documented the parameters in the latest revision.

File: lambdas/dump-v3-account-data/index.js (hunk 1 above)
Comment (alex-chew) [reply lambdas/dump-v3-account-data/index.js]: Fixed in latest revision.

**Pull request merged**